egs/himia/wuw/ctc_tdnn/tokenizer.py

```diff
@@ -0,0 +1,101 @@
+# Copyright      2023  Xiaomi Corp.        (Author: Liyong Guo)
+#
+# See ../../LICENSE for clarification regarding multiple authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import itertools
+import torch
+
+from typing import List, Tuple
+
+
+class WakeupWordTokenizer(object):
+    def __init__(
+        self,
+        wakeup_word: str = "",
+        wakeup_word_tokens: List[int] = None,
+    ) -> None:
+        """
+        Args:
+          wakeup_word: content of positive samples.
+            A sample will be treated as a negative sample unless its context
+            is exactly the same to key_words.
+          wakeup_word_tokens: A list if int represents token ids of wakeup_word.
+            For example: the pronunciation of "你好米雅" is
+            "n i h ao m i y a".
+            Suppose we are using following lexicon:
+              blk 0
+              unk 1
+              n   2
+              i   3
+              h   4
+              ao  5
+              m   6
+              y   7
+              a   8
+            Then wakeup_word_tokens for "你好米雅" is:
+             n  i  h  ao m  i  y  a
+            [2, 3, 4, 5, 6, 3, 7, 8]
+        """
+        super().__init__()
+        assert wakeup_word is not None
+        assert wakeup_word_tokens is not None
```

Comment:
if it cannot be None, please don't use None as its default value.

Reply:
fixed. thanks.